File: 6-kyu/find-the-parity-outlier/swift/solution.swift
import Foundation

func findOutlier(_ array: [Int]) -> Int {
    let noEvens = array.filter { abs($0) % 2 == 1 }
    if noEvens.count == 1 { return noEvens[0] }
    else {
        let noOdds = array.filter { abs($0) % 2 == 0 }
        return noOdds[0]
    }
}

File: 6-kyu/find-the-parity-outlier/python/solution.py
def find_outlier(integers):
    no_evens = filter(lambda item: item % 2 == 0, integers)
    if len(no_evens) == 1: return no_evens[0]
    no_odds = filter(lambda item: item % 2 == 1, integers)
    return no_odds[0]